### PR TITLE
[CARBONDATA-2495][Doc][BloomDataMap] Add document for bloomfilter datamap

### PR DESCRIPTION
**Pull request by xuchuanyin**
add document for bloomfilter datamap

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 `NA`
 - [x] Any backward compatibility impacted?
 `NA` 
 - [x] Document update required?
 `Add document`
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
 `NA`
        - How it is tested? Please attach test report.
 `NA`
        - Is it a performance related change? Please attach the performance test report.
 `NA`
        - Any additional information to help reviewers in testing this change.
        `NA`
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
 `NA`

### CODE REVIEW

**Comment by xuchuanyin** [comment]
@jackylk @chenliang613 Can you review this?

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5003/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5980/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4822/

File: docs/datamap/bloomfilter-datamap-guide.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  SHOW DATAMAP
+  ON TABLE main_table
+  ```
+It will show all DataMaps created on main table.
+
+
+## BloomFilter DataMap Introduction
+A Bloom filter is a space-efficient probabilistic data structure that is used to test whether an element is a member of a set.
+Carbondata introduce BloomFilter as an index datamap to enhance the performance of querying with precise value.
+Internally, CarbonData maintains a BloomFilter per blocklet for each index column to indicate that whether a value of the column is in this blocklet.
```
Comment (jackylk) [inline docs/datamap/bloomfilter-datamap-guide.md]: Please give a hint what the suitable scenario it is to use this datamap, for example, for queries that do precise match on high cardinality column

Comment (xuchuanyin) [reply docs/datamap/bloomfilter-datamap-guide.md]: OK

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4883/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6042/

**Pull request closed without merging**